cpu/nrf5x_common/periph/i2c_nrf52_nrf9160.c

```diff
@@ -331,6 +339,27 @@ int i2c_read_regs(i2c_t dev, uint16_t addr, uint16_t reg,
 
 int i2c_write_bytes(i2c_t dev, uint16_t addr, const void *data, size_t len,
                     uint8_t flags)
+{
+    /* These are critical for the memcpy; direct_i2c_write_bytes makes some
```

Comment:
How about a fast path

```C
if ((uintptr_t)data >= RAM_START_ADDR) {
    return direct_i2c_write_bytes(addr, data, len, flags);
}
```

Reply:
That'd be relying on an [undocumented](https://duckduckgo.com/?q=%22RAM_START_ADDR%22+site%3Adoc.riot-os.org&kt=h&ia=web) constant.

On the nRF52840, both code and XPI (memory mapped external QSPI) are all below RAM_START_ADDR, but the nRF9160 [memory map](https://infocenter.nordicsemi.com/topic/ps_nrf9160/memory.html) has a lot of stuff below; is there a RAM_END_ADDR too?

Reply:
No but you have RAM_LEN IIRC.

Reply:
None of these symbols even exist. What I got to work is

```patch
+    extern char *_ram_start_addr;
+    extern unsigned int _ram_length;
+    if ((char*)data >= _ram_start_addr && (char*)data < _ram_start_addr + _ram_length) {
+        return direct_i2c_write_bytes(dev, addr, data, len, flags);
+    }
+
```

-- that is reaching more deeply into linker stuff that I'm usually comfortable doing, but if it's what you two prefer, so be it.

Reply:
> None of these symbols even exist.

huh, we have 

```
cpu/nrf5x_common/Makefile.include
4:RAM_START_ADDR ?= 0x20000000
```
(actually could move that to `cpu/cortexm_common`) and 

```
makefiles/cflags.inc.mk
124:ifneq (,$(RAM_START_ADDR))
125:  CFLAGS += -DCPU_RAM_BASE=$(RAM_START_ADDR)
```

Reply:
Thanks for the symbol name, that indeed works without externs.

Up to today it was unlikely that someone would send data from flash, and that that someone just spent over a day hunting down three different subtle bugs that all just started showing when an innocent change in a driver that was already well tested caused trouble. I don't wish that same trouble on the first person who uses nRF9610 to send valid data from one place to another.

I've updated this PR to do the two-sided check, please review.